sdk/metric/cache_test.go

```diff
@@ -59,18 +58,5 @@ func TestCacheConcurrentSafe(t *testing.T) {
 		}(n)
 	}
 
-	done := make(chan struct{})
-	go func() {
-		wg.Wait()
-		close(done)
-	}()
-
-	assert.Eventually(t, func() bool {
-		select {
-		case <-done:
-			return true
-		default:
-			return false
-		}
-	}, timeoutSec*time.Second, 10*time.Millisecond)
+	wg.Wait()
```

Comment:
This will panic the whole test run at the test timeout instead of just failing the this test after 5 seconds. I don't see that as an improvement in behavior.

Reply:
I felt that this is a little YAGNI especially that no other `ConcurrentSafe` tests has this feature. Do you prefer reverting these changes?

Reply:
> Do you prefer reverting these changes?

yes

Reply:
Addressed https://github.com/open-telemetry/opentelemetry-go/pull/4352/commits/864b1346a54b56c3089624b30f941d8bdb864a37